File: Problems/#49/Solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        map = defaultdict(list)
        for word in strs:
            sorted_word = "".join(sorted(word))
            map[sorted_word].append(word)
        ans = list(map.values())
        return ans

File: Problems/#49/Solution.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> map = new HashMap<>();
        for (String word : strs) {
            char[] arr = word.toCharArray();
            Arrays.sort(arr);
            String sortedWord = new String(arr); 
            if (!map.containsKey(sortedWord)) map.put(sortedWord, new ArrayList<>());
            map.get(sortedWord).add(word);
        }
        List<List<String>> ans = new ArrayList<>();
        for (Map.Entry<String, List<String>> entry : map.entrySet()) ans.add(entry.getValue());
        return ans;
    }
}
